src/Sdk/WebApi/WebApi/BrokerHttpClient.cs

```diff
@@ -115,6 +115,7 @@ public async Task<TaskAgentMessage> GetRunnerMessageAsync(
             {
                 switch (brokerError.ErrorKind)
                 {
+                    case BrokerErrorKind.RunnerNotFound:
```

Comment:
I wouldn't re-use the same AccessDeniedException with ErrorCode=1. That means the runner deprecation error in a few other places. Can we throw another error here instead? I don't really care what, but probably should be different than the Runner version being too old?

Reply:
Oh interesting. I thought this was a generic mechanism to bubble out. Will look again.